crates/core/tedge_agent/src/file_transfer_server/actor.rs

```diff
@@ -141,31 +144,26 @@ mod tests {
     }
 
     #[tokio::test]
-    #[serial_test::serial]
-    async fn check_server_does_not_panic_when_port_is_in_use() -> Result<(), anyhow::Error> {
+    async fn check_server_does_not_panic_when_port_is_in_use() -> anyhow::Result<()> {
```

Comment:
This refactoring is incidental, I noticed the test had both a hard sleep (so it was comparitively slow) and not terribly deterministic (no guarantee server one would attempt to bind first, no guarantee the port binding would fail within 100ms).